atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -368,6 +368,116 @@ void WebFrame::ClearCache(v8::Isolate* isolate) {
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::Parent() const {
+  blink::WebLocalFrame* local_frame =
+    (blink::WebLocalFrame*) WebFrame::web_frame_->Parent();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::Top() const {
+  blink::WebLocalFrame* local_frame =
+    (blink::WebLocalFrame*) WebFrame::web_frame_->Top();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::FirstChild() const {
+  blink::WebLocalFrame* local_frame =
+    (blink::WebLocalFrame*) WebFrame::web_frame_->FirstChild();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::NextSibling() const {
+  blink::WebLocalFrame* local_frame =
+    (blink::WebLocalFrame*) WebFrame::web_frame_->NextSibling();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::TraverseNext() const {
+  blink::WebLocalFrame* local_frame =
+    (blink::WebLocalFrame*) WebFrame::web_frame_->TraverseNext();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::GetFrameForSelector(
+  v8::Local<v8::Value> selector_value,
+  mate::Arguments* args
+) const {
+  if (!selector_value->IsString()) {
+    args->ThrowError("selector string required");
+    return v8::Null(isolate());
+  }
+  v8::Local<v8::String> selector_v8_string = selector_value->ToString();
+  int selector_utf8_size = selector_v8_string->Utf8Length();
+  char* selector_utf8_buff = new char[selector_utf8_size];
+  selector_v8_string->WriteUtf8(selector_utf8_buff, selector_utf8_size);
+  blink::WebString selector_web_string = blink::WebString::FromUTF8(
+    selector_utf8_buff,
+    selector_utf8_size);
+  delete[] selector_utf8_buff;
+  blink::WebElement element = web_frame_->GetDocument()
+    .QuerySelector(selector_web_string);
+  blink::WebLocalFrame* element_frame =
+    blink::WebLocalFrame::FromFrameOwnerElement(element);
+  if (element_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), element_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::FindFrameByName(v8::Local<v8::Value> name_value,
```

Comment:
You can extract the name or selector values directly from `mate::Arguments` which performs the right string conversion.

```
v8::Local<v8::Value> WebFrame::FindFrameByName(mate::Arguments* args) {
  std::string name;
  if (args && args->Length() == 1 && args->GetNext(&name) {}
}
```